casper/safety_oracles/clique_oracle.py

```diff
@@ -112,3 +112,72 @@ def check_estimate_safety(self):
         # Return the number of faults we can tolerate, which is one less
         # than the number that need to equivocate.
         return fault_tolerance, len(equivocating) - 1
+
+    def check_heuristic_min_bound(self):
+        """Returns a minimum bound clique using a greedy search lookup."""
+
+        # collect edges and create graph
+        edges = self._collect_edges()
+        graph = nx.Graph()
+        graph.add_edges_from(edges)
+        nodes = graph.nodes()
+
+        # if graph is empty, break out of function
+        if not nodes:
+            print('### graph is empty, not running clique search')
+            return
+
+        # base case: find largest degree node in graph
+        max_degree_node = self.find_max_degree_node(graph, nodes)
+        if not max_degree_node:
+            raise Exception("### Did not find a node in the graph, no common estimates")
```

Comment:
Does this ever trigger? It seems like if we have any nodes (as nodes are "inferred" from edges), we should have some node with a max degree.